server.js

```diff
@@ -1185,6 +1186,10 @@ function objectBeatSender(PORT, thisId, thisIp, oneTimeOnly) {
         oneTimeOnly = false;
     }
 
+    if (typeof immediate === 'undefined') {
+        immediate = false;
+    }
```

Comment:
I think you can give the parameter a default value in the function signature instead of handling it here

`function objectBeatSender(PORT, thisId, thisIp, oneTimeOnly, immediate = false) { ... `

Reply:
Thanks for the review. I defined default values for both oneTimeOnly and immediate, so it is more modern JS.

Reply:
Thanks for bringing it up to date!